contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -47,448 +43,251 @@ contract FraudProofFacet is StateChannelCommon {
     }
 
     // ******************************* FRAUD PROOF IMPLEMENTATION *******************************
-    
-    function _handleBlockDoubleSign(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal returns (address) {
-        // Implementation for handling BlockDoubleSign proof
-        return address(0); // Replace with actual logic
+
+
+    // ------------------------------- Block Fraud Proofs --------------------------------------- 
+    function _handleBlockDoubleSign(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) pure internal returns (address) {
+        BlockDoubleSignProof memory blockDoubleSignProof = abi.decode(encodedProof, (BlockDoubleSignProof));
+
+        Block memory block1 = abi.decode(blockDoubleSignProof.block1.encodedBlock, (Block));
+        Block memory block2 = abi.decode(blockDoubleSignProof.block2.encodedBlock, (Block));
+
+        if(fraudProofVerificationContext.channelId != block1.transaction.header.channelId || fraudProofVerificationContext.channelId != block2.transaction.header.channelId) {
+            return address(0);
```

Comment:
throw an error on failure so the low level function call (delegatecall) will catch it - so we should never return address(0)

Reply:
But still better to throw a specific error so the client can catch the specific error and interpret it (e.g. missing commitment [channelId][adr][fork][height])

Reply:
so there are conflicting views
1. If we return address(0), we already handle it in verifyfraudProof
2. if we return an error we have to specify error types so that the TS client can handle specific errors.. but that can grow so much as number of errors. (is it sustainable)

Reply:
In general we have many custom errors that don't need any special handling, but help us understand what failed - The few errors that need custom handling are related to commitments and notify the client to try and fetch those commitments on-chain to reconstruct the state/environment off-chain. Everything else is just relevant for logs and debugging and doesn't need custom behavior